src/client/pages/Offer/Introduction/DetailsModal/Details/components/DetailInput.tsx

```diff
@@ -335,56 +337,53 @@ export const ExtraBuildingsInput: React.FC<ExtraBuildingsInputProps> = ({
-                      label: getExtraBuilding(value),
-                      value,
-                    })),
-                  }}
-                  formikProps={formikProps}
-                />
-                <AreaInput
-                  name={`data.extraBuildings.${index}.area`}
-                  label="DETAILS_MODULE_EXTRABUILDINGS_TABLE_SIZE_CELL_LABEL_HOUSE"
-                  formikProps={formikProps}
-                />
-                <BooleanInput
-                  name={`data.extraBuildings.${index}.hasWaterConnected`}
-                  label="DETAILS_MODULE_EXTRABUILDINGS_TABLE_WATER_CELL_LABEL_HOUSE"
-                  formikProps={formikProps}
-                />
-                <InputGroupDeleteButton
-                  type="button"
-                  onClick={() => {
-                    const isLastItemLeft =
-                      formikProps.values.data.extraBuildings?.length === 1
+          {extraBuildings?.map((_: any, index: number) => (
+            <InputGroup key={index}>
+              <DetailInput
+                name={`data.extraBuildings.${index}.type`}
+                field={{
+                  label:
+                    'DETAILS_MODULE_EXTRABUILDINGS_TABLE_BUILDINGTYPE_CELL_LABEL_HOUSE',
+                  placeholder: '',
+                  options: Object.values(ExtraBuildingType).map((value) => ({
+                    label: getExtraBuilding(value),
+                    value,
+                  })),
+                }}
+                formikProps={formikProps}
+              />
+              <AreaInput
+                name={`data.extraBuildings.${index}.area`}
+                label="DETAILS_MODULE_EXTRABUILDINGS_TABLE_SIZE_CELL_LABEL_HOUSE"
+                formikProps={formikProps}
+              />
+              <BooleanInput
+                name={`data.extraBuildings.${index}.hasWaterConnected`}
+                label="DETAILS_MODULE_EXTRABUILDINGS_TABLE_WATER_CELL_LABEL_HOUSE"
+                formikProps={formikProps}
+              />
+              <InputGroupDeleteButton
+                type="button"
+                onClick={() => {
+                  const isLastItemLeft = extraBuildings?.length === 1
 
-                    arrayHelpers.remove(index)
+                  arrayHelpers.remove(index)
 
-                    if (isLastItemLeft) {
-                      formikProps.setValues({
-                        ...formikProps.values,
-                        data: {
-                          ...formikProps.values.data,
-                          extraBuildings: [],
-                        },
-                      })
-                    }
-                  }}
-                >
-                  {textKeys.DETAILS_MODULE_EXTRABUILDINGS_TABLE_REMOVE_BUILDING_BUTTON()}
-                </InputGroupDeleteButton>
-              </InputGroup>
-            ),
-          )}
+                  if (isLastItemLeft) {
```

Comment:
Sorry for coming back to an old PR after so long, but looking at this `isLastItemLeft` - what's the point? We have removed all items in the array - so why `setValues` to an empty array instead of just leaving it to `remove()`? I know it was in the code before as well, just thought that you might know?

Reply:
We could absolutely do a rafactor on this -- I don't have any clue why this was introduced 😅